### PR TITLE
Updated Zoedepth model card

### PR DESCRIPTION
**Pull request by miniMaddy**
# What does this PR do?

Updated zoedepth model card
Fixes #36979 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Models:
- vision models: @amyeroberts 

Documentation: @stevhliu

### CODE REVIEW

**Comment by 1himan** [comment]
how do you write all of that documentation? I mean do you run and test the model on your machine and then list out the features it has, write out its intro and rest of the detes or You just take the help of other resources already available online or a mix of both. Do you use chatbots(like gpt, or cluade) to help you with it? I'm really really new to all of this could you please tell me what should I do in order to start contributing?

**Comment by miniMaddy** [comment]
Hi @1himan , I mostly used the template provided in the issue and the data that are already there in the old zoedepth model card to generate the new file. I took inspiration from vit.md file wherever I was confused about the new template. I didn't test and run the model on my local system. I looked into the documentation to write the new demo code and mostly it's old code itself. 

I didn't use any chatbot here as they might not have context over the hf documentation but I am planning to do that in the next few model card updates, after I get sense of what is expected from this pr.

I am new too. This is my first PR to hugging face. Just read the instructions and other merged/open PRs for this issue to get a sense of what is changing and made some changes. I am sure there may be mistakes and the maintainers will help me fix that. So don't take too much stress. Hope that answers your questions.

Thank You

File: docs/source/en/model_doc/zoedepth.md [hunk 1]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+[Zoedepth](https://huggingface.co/papers/2302.12288) is a metric (also called absolute) depth estimation model to generate depth maps directly from images. It gives absolute metric depth in real-world metres, instead of relative depth. ZoeDepth is pre-trained on 12 datasets using relative depth and fine-tuned on two domains (NYU and KITTI) using metric depth. A lightweight head is used with a novel bin adjustment design called metric bins module for each domain. During inference, each input image is automatically routed to the appropriate head using a latent classifier.
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:26]:
```suggestion
[ZoeDepth](https://huggingface.co/papers/2302.12288) is a depth estimation model that combines the generalization performance of relative depth estimation (how far objects are from each other) and metric depth estimation (precise depth measurement on metric scale) from a single image. It is pre-trained on 12 datasets using relative depth and 2 datasets (NYU Depth v2 and KITTI) for metric accuracy. A lightweight head with a metric bin module for each domain is used, and during inference, it automatically selects the appropriate head for each input image with a latent classifier.
```

File: docs/source/en/model_doc/zoedepth.md [hunk 2]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
 
-The abstract from the paper is the following:
+You can find all the original Zoedepth checkpoints under the [Zoedepth](https://huggingface.co/Intel?search=zoedepth) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:28]:
```suggestion
You can find all the original ZoeDepth checkpoints under the [Intel](https://huggingface.co/Intel?search=zoedepth) organization.
```

File: docs/source/en/model_doc/zoedepth.md [hunk 3]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:30]:
```suggestion
The example below demonstrates how to estimate depth with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/zoedepth.md [hunk 4]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/zoedepth_architecture_bis.png"
-alt="drawing" width="600"/>
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-<small> ZoeDepth architecture. Taken from the <a href="https://arxiv.org/abs/2302.12288">original paper.</a> </small>
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:36]: Update slightly with:

```py
import requests
import torch
from transformers import pipeline
from PIL import Image

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)
pipeline = pipeline(
    task="depth-estimation",
    model="Intel/zoedepth-nyu-kitti",
    torch_dtype=torch.float16,
    device=0
)
results = pipeline(image)
results["depth"]
```

File: docs/source/en/model_doc/zoedepth.md [hunk 5]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+import requests
+from PIL import Image
+from transformers import AutoModelForDepthEstimation, AutoImageProcessor
+
+image_processor = AutoImageProcessor.from_pretrained(
+    "Intel/zoedepth-nyu-kitti"
+)
+model = AutoModelForDepthEstimation.from_pretrained(
+    "Intel/zoedepth-nyu-kitti",
+    device_map=0
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:65]:
```suggestion
    device_map="auto"
```

File: docs/source/en/model_doc/zoedepth.md [hunk 6]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+)
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:67]:
```suggestion
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
```

File: docs/source/en/model_doc/zoedepth.md [hunk 7]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+image = Image.open(requests.get(url, stream=True).raw)
+inputs = image_processor(image, return_tensors="pt").to("cuda")
+
+with torch.no_grad():
+  outputs = model(inputs)
+
+# interpolate to original size and visualize the prediction
+## ZoeDepth dynamically pads the input image. Thus we pass the original image size as argument
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:75]:
```suggestion
## ZoeDepth dynamically pads the input image, so pass the original image size as argument
```

File: docs/source/en/model_doc/zoedepth.md [hunk 8]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+## to `post_process_depth_estimation` to remove the padding and resize to original dimensions.
+post_processed_output = image_processor.post_process_depth_estimation(
+    outputs,
+    source_sizes=[(image.height, image.width)],
+)
+
+predicted_depth = post_processed_output[0]["predicted_depth"]
+depth = (predicted_depth - predicted_depth.min()) / (predicted_depth.max() - predicted_depth.min())
+depth = depth.detach().cpu().numpy() * 255
+depth = Image.fromarray(depth.astype("uint8"))
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:85]:
```suggestion
Image.fromarray(depth.astype("uint8"))
```

File: docs/source/en/model_doc/zoedepth.md [hunk 9]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
-</code></pre>
-</Tip>
+</hfoption>
+</hfoptions>
 
-## Resources
+## Notes
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with ZoeDepth.
+- In the [original implementation](https://github.com/isl-org/ZoeDepth/blob/edb6daf45458569e24f50250ef1ed08c015f17a7/zoedepth/models/depth_model.py#L131) ZoeDepth model performs inference on both the original and flipped images and averages out the results. The ```post_process_depth_estimation``` function can handle this for us by passing the flipped outputs to the optional ```outputs_flipped``` argument:
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:93]:
```suggestion
- In the [original implementation](https://github.com/isl-org/ZoeDepth/blob/edb6daf45458569e24f50250ef1ed08c015f17a7/zoedepth/models/depth_model.py#L131) ZoeDepth performs inference on both the original and flipped images and averages the results. The `post_process_depth_estimation` function handles this by passing the flipped outputs to the optional `outputs_flipped` argument as shown below.
```

File: docs/source/en/model_doc/zoedepth.md [hunk 10]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/zoedepth_architecture_bis.png"
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:31]: Let's actually keep the image.

File: docs/source/en/model_doc/zoedepth.md [hunk 11]
```diff
@@ -14,100 +14,95 @@ rendered properly in your Markdown viewer.
...
+    with torch.no_grad():
+        outputs = model(pixel_values)
+        outputs_flipped = model(pixel_values=torch.flip(inputs.pixel_values, dims=[3]))
+        post_processed_output = image_processor.post_process_depth_estimation(
+            outputs,
+            source_sizes=[(image.height, image.width)],
+            outputs_flipped=outputs_flipped,
+        )
+   ```
+- A demo notebook regarding inference with ZoeDepth models can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/ZoeDepth).
```
Comment (stevhliu) [inline docs/source/en/model_doc/zoedepth.md:104]: We're also going to keep the `## Resources` section moving forward, sorry! 

```suggestion
## Resources
- Refer to this [notebook](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/ZoeDepth) for an inference example.
```

**Review by stevhliu — COMMENTED** [review]
Very nice work, thanks!

**Comment by miniMaddy** [comment]
Hi @stevhliu , I completed the suggested changes. Please review it. Thanks

**Review by stevhliu — APPROVED** [review]
Looks great, thanks! 🤗
